air-quality-backend/etl/tests/forecast/forecast_texture_storer_test.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+from unittest.mock import patch
+
+from etl.src.forecast.forecast_texture_storer import (
+    _chunk_data_array,
+    _create_output_directory,
+    _write_texture_to_disk,
+    save_data_textures,
+)
+from shared.tests.util.mock_forecast_data import default_time
+
+
+@pytest.mark.parametrize(
+    "num_time_steps, expected_num_chunks",
+    [
+        (4, 3),  # Standard case: 3 chunks
+        (11, 1),  # Edge case: Only 1 chunk, as num_time_steps equals total time steps
+        (1, 11),  # Edge case: Each time step is its own chunk
+        (0, 1),  # Edge case: num_time_steps is 0, should default to 1 chunk
+        (-1, 1),  # Edge case: num_time_steps is negative, should default to 1 chunk
+        (
+            20,
+            1,
+        ),  # Edge case: should default to 1 chunk
+    ],
+)
+def test__chunk_data_array(num_time_steps, expected_num_chunks):
+    rgb_data_array = np.random.rand(11, 121, 1).astype(np.uint8)
+    num_lon = 11
+    time_vector = xr.DataArray(np.linspace(0, 10, 11), dims="time")
+
+    chunks, time_steps_dict = _chunk_data_array(
+        rgb_data_array, num_time_steps, num_lon, time_vector
+    )
+
+    assert len(chunks) == expected_num_chunks
+    assert all(chunk.shape[0] == 11 for chunk in chunks)
+    assert all(chunk.shape[2] == 1 for chunk in chunks)
+    assert len(time_steps_dict) == expected_num_chunks
+
+    for i in range(expected_num_chunks):
+        assert "time_start" in time_steps_dict[i]
+        assert "time_end" in time_steps_dict[i]
+
+
+@pytest.fixture
+def mock_getcwd():
+    with patch("os.getcwd", return_value="/current/working/directory"):
+        yield
+
+
+@pytest.fixture
+def mock_makedirs():
+    with patch("os.makedirs") as mock:
+        yield mock
+
+
+@patch("os.makedirs")
+@patch("os.getcwd", return_value="/mock/current/directory")
+@patch("os.path.exists", side_effect=lambda path: path.startswith("/app/data_textures"))
+def test__create_output_directory(mock_exists, mock_getcwd, mock_makedirs):
+    forecast_date = "2024-07-09"
+
+    # Test case where the directory does not exist in /app/data_textures
+    mock_exists.side_effect = lambda path: False
+    expected_directory = f"/mock/current/directory/data_textures/{forecast_date}"
+    result = _create_output_directory(forecast_date)
+    mock_makedirs.assert_called_once_with(expected_directory, exist_ok=True)
+    assert result == expected_directory
+
+    # Reset mocks for the next test case
+    mock_makedirs.reset_mock()
+    mock_exists.side_effect = lambda path: path == f"/app/data_textures/{forecast_date}"
+
+    # Test case where the directory exists in /app/data_textures
+    expected_directory = f"/app/data_textures/{forecast_date}"
+    result = _create_output_directory(forecast_date)
+    mock_makedirs.assert_called_once_with(expected_directory, exist_ok=True)
+    assert result == expected_directory
+
+
+@patch("PIL.Image.Image.save")
+def test__save_data_texture(mock_save):
```

Comment:
Should be called test__write_texture_to_disk ?

Reply:
Yepp changed.